components/Post.php

```diff
@@ -103,6 +103,14 @@ protected function loadPost()
                 $category->setUrl($this->categoryPage, $this->controller);
             });
         }
+        
+        /*
+         * Throw a 404 error when a blogpost does not exist
+         */
+        if (!$post || !$post->exists) {
+            $this->setStatusCode(404);
+            return $this->controller->run('404');
+        }
```

Comment:
@maartenmachiels The block of code above should be run immediately after trying to retrieve the post. On this line here (https://github.com/rainlab/blog-plugin/blob/0b4ef4609fd06f8b83d9beca8149d2cd32cc5850/components/Post.php#L96), you could change the call to `firstOrFail()` and wrap the line in a `try..catch` block. If a `Illuminate\Database\Eloquent\ModelNotFoundException` exception is caught, you can then run your 404 code.